src/librbd/ImageCtx.cc

```diff
@@ -850,7 +850,11 @@ struct C_InvalidateCache : public Context {
     on_safe = util::create_async_context_callback(*this, on_safe);
 
     assert(owner_lock.is_locked());
-    if (object_cacher != NULL) {
+
+    bool journaling = false;
+    journaling = (journal != NULL && !journal->is_journal_replaying());
+
+    if (!journaling && object_cacher != NULL) {
```

Comment:
@dillaman thanks for the review! Do you mean we should move the C_FlushCache stuff to AioImageRequest<flush>::send_request() ? To my understanding this only handles the librbd::aio_flush() API, how about the librbd::flush()?